File: Resueltos_por_competidor/Tomás_Triana_Galvis/ConcatenationOfLanguages.cpp
#include<bits/stdc++.h>
typedef long long ll;
using namespace std;
/*
 * Autor: Tomás Triana Galvis
 * Problema: Concatenation of Languages
 * Juez online: UVA 10887
 * Veredicto: Accepted
 * Url: https://onlinejudge.org/index.php?option=com_onlinejudge&Itemid=8&page=show_problem&problem=1828
 **/ 

int main(){
    ll t,n,m,ind =1;
    string a;
    unordered_set<string> ans;
    cin>>t;
    cin.ignore();
    while(t--){
        ans.clear();
        cin>>n>>m;
        vector<string> arr1(n),arr2(m);
        cin.ignore();
        
        for(int i =0;i<n;i++){
            getline(cin,a);
            arr1[i] = a;
        }
        for(int i=0;i<m;i++){
            getline(cin,a);
            arr2[i] = a;
        }
        for(int i=0;i<n;i++){
            for(int j=0;j<m;j++){
                
                ans.insert(arr1[i]+arr2[j]);
            }
        }
        cout<<"Case "<<ind<<": "<<ans.size()<<"\n";
        ind++;
    }

    return 0;
}

File: Resueltos_por_competidor/Mariana_Pineda/ConcatenationOfLanguages.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.HashSet;
import java.util.StringTokenizer;

public class Main {
	/*
	 * Autor: mariana 
	 * Problema : 10887-Concatenation of Languages
	 * Juez online: OnlineJudge
	 * Veredicto: Accepted 
	 * URL: https://onlinejudge.org/index.php?option=com_onlinejudge&Itemid=8&page=show_problem&problem=1828
	 */
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int t = Integer.parseInt(br.readLine());
		int m = 0;
		int n = 0;
		int a = 1;
		HashSet<String> ay = new HashSet<>();
		StringTokenizer st;
		String[] ayuda, help;
		while (t > 0) {
			st = new StringTokenizer(br.readLine());
			m = Integer.parseInt(st.nextToken());
			n = Integer.parseInt(st.nextToken());
			ayuda = new String[m];
			help = new String[n];
			for (int i = 0; i < m; i++) {
				ayuda[i] = br.readLine();
			}
			for (int i = 0; i < n; i++) {
				help[i] = br.readLine();
			}

			for (int i = 0; i < ayuda.length; i++) {
				for (int j = 0; j < help.length; j++) {
					ay.add(ayuda[i] + help[j]);
				}
			}
			System.out.println("Case " + a + ": " + ay.size());
			ay.clear();
			a++;
			t--;
		}

	}
}
